lib/adapters/idb/idb.js

```diff
@@ -999,57 +819,87 @@ function init(api, opts, callback) {
 
     idb.onversionchange = function () {
       idb.close();
-      delete cachedDBs[name];
+      delete cachedDBs[dbName];
     };
     idb.onabort = function () {
       idb.close();
-      delete cachedDBs[name];
+      delete cachedDBs[dbName];
     };
 
-    var txn = idb.transaction([META_STORE, DETECT_BLOB_SUPPORT_STORE],
-      'readwrite');
+    var txn = idb.transaction([
+        META_STORE,
+        DETECT_BLOB_SUPPORT_STORE,
+        DOC_STORE
+      ], 'readwrite');
 
     var req = txn.objectStore(META_STORE).get(META_STORE);
 
+    var blobSupport = null;
+    var docCount = null;
+    var instanceId = null;
+
     req.onsuccess = function (e) {
 
       var checkSetupComplete = function () {
-        if (api._blobSupport === null || !idStored) {
+        if (blobSupport === null || docCount === null ||
+            instanceId === null) {
           return;
         } else {
-          cachedDBs[name] = {
-            idb: idb,
+          api._meta = {
+            name: dbName,
             instanceId: instanceId,
-            blobSupport: api._blobSupport,
-            loaded: true
+            blobSupport: blobSupport,
+            docCount: docCount
+          };
+
+          cachedDBs[dbName] = {
+            idb: idb,
+            global: api._meta
           };
           callback(null, api);
         }
       };
 
+      //
+      // fetch/store the id
+      //
+
       var meta = e.target.result || {id: META_STORE};
-      if (name  + '_id' in meta) {
-        instanceId = meta[name + '_id'];
-        idStored = true;
+      if (dbName  + '_id' in meta) {
+        instanceId = meta[dbName + '_id'];
         checkSetupComplete();
       } else {
         instanceId = utils.uuid();
-        meta[name + '_id'] = instanceId;
+        meta[dbName + '_id'] = instanceId;
         txn.objectStore(META_STORE).put(meta).onsuccess = function () {
-          idStored = true;
           checkSetupComplete();
         };
       }
 
+      //
+      // check blob support
+      //
+
       if (!blobSupportPromise) {
         // make sure blob support is only checked once
         blobSupportPromise = checkBlobSupport(txn, idb);
       }
 
       blobSupportPromise.then(function (val) {
-        api._blobSupport = val;
+        blobSupport = val;
         checkSetupComplete();
       });
+
+      //
+      // count docs
+      //
+
+      var index = txn.objectStore(DOC_STORE).index('deletedOrLocal');
+      index.count(IDBKeyRange.only('0')).onsuccess = function (e) {
+        docCount = e.target.result;
+        checkSetupComplete();
+      };
+
```

Comment:
we can just count the docs once now, when the database starts up. we might also want to reconsider this `deletedOrLocal` index entirely, since this is the only place we're using it anymore IIRC.